go/plugins/compat_oai/openai/generate.go

```diff
@@ -0,0 +1,132 @@
+package openai
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/firebase/genkit/go/ai"
+	"github.com/openai/openai-go"
+)
+
+// ModelGenerator handles OpenAI generation requests
+type ModelGenerator struct {
+	client    *openai.Client
+	modelName string
+	request   *openai.ChatCompletionNewParams
+}
+
+// NewModelGenerator creates a new ModelGenerator instance
+func NewModelGenerator(client *openai.Client, modelName string) *ModelGenerator {
+	return &ModelGenerator{
+		client:    client,
+		modelName: modelName,
+		request: &openai.ChatCompletionNewParams{
+			Model: openai.F(modelName),
+		},
+	}
+}
+
+// WithMessages adds messages to the request
+func (g *ModelGenerator) WithMessages(messages []*ai.Message) *ModelGenerator {
+	oaiMessages := make([]openai.ChatCompletionMessageParamUnion, 0, len(messages))
+	for _, msg := range messages {
+		content := g.concatenateContent(msg.Content)
+		switch msg.Role {
+		case ai.RoleSystem:
+			oaiMessages = append(oaiMessages, openai.SystemMessage(content))
+		case ai.RoleModel:
+			oaiMessages = append(oaiMessages, openai.AssistantMessage(content))
+		case ai.RoleTool:
+			oaiMessages = append(oaiMessages, openai.ToolMessage("", content)) // TODO: Add tool ID if needed
+		default:
+			oaiMessages = append(oaiMessages, openai.UserMessage(content))
+		}
+	}
+	g.request.Messages = openai.F(oaiMessages)
+	return g
+}
+
+// WithConfig adds configuration parameters from the model request
+func (g *ModelGenerator) WithConfig(config any) *ModelGenerator {
+	if config != nil {
+		// TODO: Implement configuration from model request
+		// modelRequest.Config is any type
+	}
+	return g
+}
+
+// WithTools adds tools to the request
+func (g *ModelGenerator) WithTools(tools []ai.Tool, choice ai.ToolChoice) *ModelGenerator {
+	// TODO: Implement tools from model request
+	// see vertex ai recent pr here for reference: https://github.com/firebase/genkit/pull/2259
+	return g
+}
```

Comment:
Need to implement still - how did we implement a config in the js plugin @xavidop ?

Reply:
I will take care of it